data-tck/src/main/java/io/micronaut/data/tck/repositories/BookSpecifications.java

```diff
@@ -1,24 +1,14 @@
 package io.micronaut.data.tck.repositories;
 
-import io.micronaut.data.model.jpa.criteria.PersistentEntityFrom;
 import io.micronaut.data.repository.jpa.criteria.PredicateSpecification;
 import io.micronaut.data.tck.entities.Book;
 
 public class BookSpecifications {
 
-    public static PredicateSpecification<Book> titleEquals(String title) {
-        return (root, criteriaBuilder) ->  {
-            ((PersistentEntityFrom) root).join("genre", io.micronaut.data.annotation.Join.Type.LEFT_FETCH);
-            return criteriaBuilder.equal(root.get("title"), title);
-        };
-    }
+    private BookSpecifications() {}
 
-    public static PredicateSpecification<Book> justJoin() {
-        return (root, criteriaBuilder) -> {
-            ((PersistentEntityFrom) root).join("genre", io.micronaut.data.annotation.Join.Type.LEFT_FETCH);
-            return null;
-        };
+    public static PredicateSpecification<Book> titleEquals(String title) {
```

Comment:
Can you please also add a test where the join is used on the method and in the criteria.

Reply:
Added